src/ini/config.c

```diff
@@ -0,0 +1,79 @@
+#include "config.h"
+#include "ini.h"
+#include <string.h>
+
+int match(const char *section, const char *check_section, const char *name, const char *check_name) {
+        if (strcmp(section, check_section) == 0 && strcmp(name, check_name) == 0) {
```

Comment:
If ```section``` is NULL, the program will seg fault.  Each function argument with pointer type needs a NULL pointer check.  If the pointers are NULL, an error should be logged.  You can also use ```assert()```:
```assert(section != NULL);```

Reply:
ACK